blob/repro_test.go

```diff
@@ -0,0 +1,54 @@
+package blob_test
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v2/pkg/proof"
+	"github.com/celestiaorg/celestia-node/blob"
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/nmt/pb"
+)
+
+// Reported at https://github.com/celestiaorg/celestia-node/issues/3731.
+func TestCommitmentProofRowProofVerifyWithEmptyRoot(t *testing.T) {
+	cp := &blob.CommitmentProof{
+		RowProof: proof.RowProof{
+			Proofs: []*proof.Proof{{}},
+		},
+	}
+	root := []byte{0xd3, 0x4d, 0x34}
+	_, _ = cp.Verify(root, 1)
```

Comment:
```suggestion
	_, err := cp.Verify(root, 1)
	assert.NoError(t, err)
```
maybe add an assertion in all of them to make sure we're receiving an error

Reply:
Initially it was to ensure no crashes but sure, I've added a check to ensure a non-nil error is returned for each.